workflow/metrics/metrics.go

```diff
@@ -0,0 +1,145 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/argoproj/argo/pkg/apis/workflow"
+	"github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	"github.com/argoproj/argo/workflow/common"
+)
+
+const (
+	argoNamespace      = "argo"
+	workflowsSubsystem = "workflows"
+)
+
+type ServerConfig struct {
+	Path string
+	Port string
+	TTL  time.Duration
+}
+
+type Metrics struct {
+	registry     *prometheus.Registry
+	serverConfig ServerConfig
+
+	updatesReapplied                prometheus.Counter
+	podResourceVersionRepeated      prometheus.Counter
+	podsProcessed                   prometheus.Counter
+	workflowResourceVersionRepeated prometheus.Counter
```

Comment:
can I request you remove these today - keep this PR focussed on the refactor

Reply:
Absolutely